frontend/src/features/public-form/components/FormFields/FormFields.tsx

```diff
@@ -113,14 +123,19 @@ export const FormFields = ({
         {!!formFields?.length && (
           <Box bg={'white'} py="2.5rem" px={{ base: '1rem', md: '2.5rem' }}>
             <Stack spacing="2.25rem">
-              {!isEmpty(fieldPrefillMap) && (
-                <InlineMessage variant="warning">
-                  The highlighted fields in this form have been pre-filled
-                  according to the link that you clicked. Please check that
-                  these values are what you intend to submit, and edit if
-                  necessary.
-                </InlineMessage>
-              )}
+              {
+                // Check that PrefillMap is not empty and that at least one field is not locked and has a value
+                !isEmpty(fieldPrefillMap) &&
+                  Object.values(fieldPrefillMap).some(
+                    (field) => !field.lockPrefill && field.prefillValue,
+                  ) && (
+                    <InlineMessage variant="warning">
+                      Highlighted fields below have been pre-filled according to
+                      the form link you clicked. You may edit these fields if
+                      necessary, except non-editable fields with a lock icon.
+                    </InlineMessage>
+                  )
```

Comment:
/nit: we could have more descriptive messages here to be specific in various combination of locked/not-locked pre-filled fields and their values.

Reply:
thanks @timotheeg ! this is a good idea.

@staceytan1998 please review the copy for the 3 cases: locked and normal prefill, normal prefill only, locked prefill only

**Normal and Locked prefill**
<img width="781" alt="Screenshot 2023-05-14 at 4 08 30 PM" src="https://github.com/opengovsg/FormSG/assets/63710093/86938684-2c96-450f-b71e-ac35114e74d6">

**Normal prefill only**
<img width="773" alt="Screenshot 2023-05-14 at 4 09 04 PM" src="https://github.com/opengovsg/FormSG/assets/63710093/983cc494-db14-4fa8-a2c3-e96d6bbcd284">

**Locked prefill only**
<img width="759" alt="Screenshot 2023-05-14 at 4 09 41 PM" src="https://github.com/opengovsg/FormSG/assets/63710093/2efd4e73-c50c-4d05-8118-1ad18a74284a">

Reply:
For locked prefill only, we can omit the words 'with a lock icon'! So it will be 'Highlighted fields below have been pre-filled according to the form link you clicked. These are non-editable fields.' 

Thanks Shu Li!

Reply:
edited the copy for locked prefill only